### PR TITLE
Increased timeout in long running test FramingSpec 

### LINKED ISSUE #21673: FAILED: FramingSpec
https://jenkins.akka.io:8498/job/akka-nightly/3775/consoleFull

```
--- Stream actors debug dump ---
Stream is completed. No debug information is available
[info] - must work with various byte orders, frame lengths and offsets *** FAILED *** (5 seconds, 944 milliseconds)
[info]   java.util.concurrent.TimeoutException: Futures timed out after [3 seconds]
[info]   at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:219)
[info]   at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
[info]   at scala.concurrent.Await$$anonfun$result$1.apply(package.scala:190)
[info]   at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
[info]   at scala.concurrent.Await$.result(package.scala:190)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2$$anonfun$apply$mcV$sp$3$$anonfun$apply$mcVI$sp$3$$anonfun$apply$1$$anonfun$apply$mcVI$sp$4.apply(FramingSpec.scala:182)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2$$anonfun$apply$mcV$sp$3$$anonfun$apply$mcVI$sp$3$$anonfun$apply$1$$anonfun$apply$mcVI$sp$4.apply(FramingSpec.scala:174)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2$$anonfun$apply$mcV$sp$3$$anonfun$apply$mcVI$sp$3$$anonfun$apply$1.apply$mcVI$sp(FramingSpec.scala:174)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2$$anonfun$apply$mcV$sp$3$$anonfun$apply$mcVI$sp$3$$anonfun$apply$1.apply(FramingSpec.scala:173)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2$$anonfun$apply$mcV$sp$3$$anonfun$apply$mcVI$sp$3$$anonfun$apply$1.apply(FramingSpec.scala:173)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2$$anonfun$apply$mcV$sp$3$$anonfun$apply$mcVI$sp$3.apply(FramingSpec.scala:173)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2$$anonfun$apply$mcV$sp$3$$anonfun$apply$mcVI$sp$3.apply(FramingSpec.scala:172)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2$$anonfun$apply$mcV$sp$3.apply$mcVI$sp(FramingSpec.scala:172)
[info]   at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2.apply$mcV$sp(FramingSpec.scala:171)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2.apply(FramingSpec.scala:171)
[info]   at akka.stream.scaladsl.FramingSpec$$anonfun$2$$anonfun$apply$mcV$sp$2.apply(FramingSpec.scala:171)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:1078)
[info]   at org.scalatest.TestSuite$class.withFixture(TestSuite.scala:196)
[info]   at akka.stream.testkit.StreamSpec.withFixture(StreamSpec.scala:26)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:1075)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
[info]   at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:1088)
[info]   at akka.testkit.AkkaSpec.runTest(AkkaSpec.scala:57)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:396)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #21673

+Boyscouting to the max: idiomatic dealing of futures in scalatest...

### CODE REVIEW

**Comment by johanandren** [comment]
BTW - that test takes 17 seconds, does it really have to?

**Comment by ktoso** [comment]
LGTM, for now good enough to address these timeouts IMO.
Can be a new ticket about making it be quicker?

**Comment by johanandren** [comment]
I'll check if the fix for #21530 maybe fixes that too, as it is about framing being slow.

**Comment by johanandren** [comment]
Yup, takes 2s with that bytestring optimization.

**Comment by ktoso** [comment]
Yeah, that would be nice if it does.

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala [hunk 1]
```diff
@@ -5,16 +5,15 @@ package akka.stream.scaladsl
...
 import java.util.concurrent.ThreadLocalRandom
-import java.util.concurrent.atomic.AtomicLong
 
+import akka.stream._
 import akka.stream.scaladsl.Framing.FramingException
 import akka.stream.stage.{ GraphStage, _ }
 import akka.stream.testkit.{ StreamSpec, TestPublisher, TestSubscriber }
-import akka.stream._
 import akka.util.{ ByteString, ByteStringBuilder }
+import org.scalatest.concurrent.PatienceConfiguration.Timeout
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala]: How is this configured and what are the default values?
Do I read it correctly that the default timeout is 150 ms, and the retry interval is 15 ms?

I'm not sure if it's good to start mixing this with the testkit way of doing it. Apart from that it's cool to use latest scalatest features, what is the advantage of using this?

Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala]: `implicit val patience = PatienceConfig(testKitSettings.DefaultTimeout.duration)` in `AkkaSpec`

Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala]: ok, shouldn't the retry timeout be increased also? 15 ms might add too much load. In testkit we use 100 ms as default for `awaitAssert`

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala (hunk 1 above)
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala]: True. I'll fix that.

**Comment by ktoso** [comment]
LGTM

**Pull request merged**